RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -428,39 +623,61 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
   // get total charge in calibration channels
   if(hCalib.isValid() == true)
   {
+
+     // store the data from the lasermon fibers
+     std::map<int, std::vector<int> > lasmon_adcs;
+     std::map<int, std::vector<int> > lasmon_capids;
+
+     // we may find the fibers in different orders, initialize them here
+     if( fillLaserMonitor_ ) {
+         for( unsigned i = 0; i < laserMonDetTypeList_.size() ; ++i ) {
+             lasmon_adcs[i] = std::vector<int>();
+             lasmon_capids[i] = std::vector<int>();
+         }
+     }
+
      for(HcalCalibDigiCollection::const_iterator digi = hCalib->begin(); digi != hCalib->end(); digi++)
      {
         if(digi->id().hcalSubdet() == 0)
            continue;
 
-        /*
-        HcalCalibDetId cell = digi->id();
-        DetId detcell = (DetId)cell;
-        
-        const HcalChannelStatus* mydigistatus=myqual->getValues(detcell.rawId());
-
-        if(mySeverity->dropChannel(mydigistatus->getValue()))
-           continue;
-        if(digi->zsMarkAndPass())
-           continue;
+        // Fill the lasermonitor channels
+        if( fillLaserMonitor_ ) {
+          int dettype = digi->id().hcalSubdet( );
+          int iphi    = digi->id().iphi();
+          int ieta    = digi->id().ieta();
+         
+          // check that we have the correct dettype
+          if (std::find( laserMonDetTypeList_.begin(), laserMonDetTypeList_.end(),
+                         dettype ) != laserMonDetTypeList_.end() ) {
+              // check that we have a contained IPhi
+            if( std::find( laserMonIPhiList_.begin(), laserMonIPhiList_.end(),
+                           iphi ) != laserMonIPhiList_.end() ) {
+              // check that we have a contained IEta
+              if( std::find( laserMonIEtaList_.begin(), laserMonIEtaList_.end(),
+                             ieta ) != laserMonIEtaList_.end() ) {
+                // we have a lasmon channel, find the index in the list of inputs
+                for( unsigned idx = 0; idx < laserMonDetTypeList_.size(); ++idx ) {
+                  if( dettype == laserMonDetTypeList_[idx] &&
+                        iphi  == laserMonIPhiList_[idx] && 
+                        ieta  == laserMonIEtaList_[idx] ) {
+
+                    // now get the digis
+                    int ts_size = int(digi->size());
+                    for(int i = 0; i < ts_size; i++) {
```

Comment:
generic question - all the array indices are better kept as unsigned ints. Why make them ints?

Reply:
I switched them to unsigned